server/src/auth/auth.service.ts

```diff
@@ -126,8 +126,14 @@ export class AuthService {
       dcp_nycid_guid: GUID,
     });
 
+    const nycIdEmailRegEx = new RegExp(`^${mail}$`, 'gi');
+
     // if their e-mail is validated, associate the NYCID guid
-    if (nycExtEmailValidationFlag && contact.dcp_nycid_guid !== GUID) {
+    if (
+      nycExtEmailValidationFlag
+      && contact.dcp_nycid_guid !== GUID
+      && contact.emailaddress1.match(nycIdEmailRegEx)
```

Comment:
Should you add a check for `givenName` and `sn` here? 
Or actually it seems like this first update can forego updating `firstname` and `lastname`, since the update below (lines 147-148) takes care of it. 

I know this came in a previous PR, but having a bit of trouble following the logic of these two conditions (line 132 and 145)